### DISCUSSION TITLE
How to handle multichannel with v0.3.0-10?

### DISCUSSION THREAD

**Discussion by hecrogu** [saleor/saleor-sdk#130] [General]
Hi,
It's a question about how can we handle the multichannel feature of Saleor in v0.3.0-10 of SDK. 
Can we change the channel configured in runtime? I didn't see any exposed method to perform this action in that version. I'm using this version cause it's the version that the Saleor storefront is using and we are trying to handle multichannel features.

**Comment by rrrnld** [comment]
I am interested in this as well. Is there a recommended way to use the SDK across channels? I would like to write plugins and apps for Saleor using TypeScript, and if I want to perform an action for all channels I currently have to use separate client per channel if I understand this correctly. Is there a less clunky, idiomatic way to do this that I don't see?

**Reply by rrrnld** [reply]
After some digging I found this, which allows calling `setChannel` at any point, letting me cycle through channels: https://github.com/saleor/saleor-sdk/blob/343eb5648e45bca018a73341fe581d899e32d88c/src/core/createSaleorClient.ts#L18-L21
